common/wallet_tx.c

```diff
@@ -34,18 +61,21 @@ struct command_result *wtx_select_utxos(struct wallet_tx *tx,
 {
 	struct command_result *res;
 	u64 fee_estimate;
+
 	if (tx->all_funds) {
-		u64 amount;
+		struct amount_sat amount;
 		tx->utxos = wallet_select_all(tx->cmd, tx->cmd->ld->wallet,
 					      fee_rate_per_kw, out_len,
-					      &amount,
+					      &amount.satoshis,
 					      &fee_estimate);
 		res = check_amount(tx, amount);
 		if (res)
 			return res;
 
-		if (amount <= tx->amount) {
-			tx->change = 0;
+		/* tx->amount is max permissible */
+		if (amount_sat_less(amount, tx->amount)) {
```

Comment:
i think you want `less_eq` here?

Reply:
Good catch!  Effect should be the same, but it was *not* a deliberate change!